controllers/componentdetectionquery_controller.go

```diff
@@ -184,18 +195,149 @@
+											},
+											{
+												Name:  "NAMESPACE",
+												Value: req.Namespace,
+											},
+											{
+												Name:  "GITHUB_TOKEN",
+												Value: gitToken,
+											},
+											{
+												Name:  "CONTEXT_PATH",
+												Value: context,
+											},
+											{
+												Name:  "REVISION",
+												Value: revision,
+											},
+											{
+												Name:  "URL",
+												Value: source.URL,
+											},
+											{
+												Name:  "DEVFILE_REGISTRY_URL",
+												Value: r.DevfileRegistryURL,
+											},
+											{
+												Name:  "DEVFILE_PATH",
+												Value: devfilePath,
+											},
+											{
+												Name:  "DOCKERFILE_PATH",
+												Value: dockerfilePath,
+											},
+											{
+												Name:  "IS_DEVFILE_PRESENT",
+												Value: strconv.FormatBool(isDevfilePresent),
+											},
+											{
+												Name:  "IS_DOCKERFILE_PRESENT",
+												Value: strconv.FormatBool(isDockerfilePresent),
+											},
+											{
+												Name:  "CREATE_K8S_Job",
+												Value: "true",
+											},
+										},
+									},
+								},
+								RestartPolicy: corev1.RestartPolicyNever,
+							},
+						},
+						BackoffLimit: &backOffLimit,
+					},
+				}
+				err = r.Client.Create(ctx, jobSpec, &client.CreateOptions{})
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Error creating cdq analysis job %s... %v", jobName, req.NamespacedName))
+					r.SetCompleteConditionAndUpdateCR(ctx, req, &componentDetectionQuery, copiedCDQ, err)
+					return ctrl.Result{}, nil
+				} else {
+					//print job details
+					log.Info(fmt.Sprintf("Successfully created cdq analysis job %v, waiting for config map to be created... %v", jobName, req.NamespacedName))
+				}
 
-			devfilesMapReturned, devfilesURLMapReturned, dockerfileContextMapReturned, componentPortsMapReturned, branch, err := cdqanalysis.CloneAndAnalyze(k8sInfoClient, gitToken, req.Namespace, req.Name, context, devfilePath, dockerfilePath, source.URL, source.Revision, r.DevfileRegistryURL, isDevfilePresent, isDockerfilePresent)
-			componentDetectionQuery.Spec.GitSource.Revision = branch
-			if err != nil {
-				log.Error(err, fmt.Sprintf("Error running cdq analysis... %v", req.NamespacedName))
-				r.SetCompleteConditionAndUpdateCR(ctx, req, &componentDetectionQuery, copiedCDQ, err)
-				return ctrl.Result{}, nil
+				cm, err := waitForConfigMap(clientset, ctx, req.Name, req.Namespace)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Error waiting for configmap creation ...%v", req.NamespacedName))
+					r.SetCompleteConditionAndUpdateCR(ctx, req, &componentDetectionQuery, copiedCDQ, err)
+					cleanupK8sResources(log, clientset, ctx, fmt.Sprintf("%s-job", req.Name), req.Name, req.Namespace)
+					return ctrl.Result{}, nil
+				}
+				var errMapReturned map[string]string
+				json.Unmarshal(cm.BinaryData["devfilesMap"], &devfilesMapReturned)
```

Comment:
Can we handle all json unmarshalling errors?

Reply:
we are not expecting to receive all those data. if any map is empty or the returnErr map is nil, it will result in unmarshal error as well. So I can only log those error instead of return Reconcile immediately
```
"5 errors occurred:\n\t* unmarshal devfilesMap: unexpected end of JSON input\n\t* unmarshal dockerfileContextMap: unexpected end of JSON input\n\t* unmarshal devfilesURLMap: unexpected end of JSON input\n\t* unmarshal componentPortsMap: unexpected end of JSON input\n\t* unmarshal revision: unexpected end of JSON input\n\n"
```